test/xcdr/xcdrv1.cpp

```diff
@@ -0,0 +1,289 @@
+    //}
+
+    //{ Decoding optional not present
+    _T dec_value {0};
+    cdr.reset();
+    cdr.read_encapsulation();
+    ASSERT_EQ(cdr.get_encoding_flag(), encoding);
+    ASSERT_EQ(cdr.endianness(), endianness);
+    cdr.deserialize_type(encoding, [&dec_value](Cdr& cdr_inner, const MemberId& mid) -> bool
+            {
+                bool ret_value = true;
+                switch (mid.id)
+                {
+                    case 1:
+                        cdr_inner.deserialize_member(dec_value);
+                        break;
+                    default:
+                        ret_value = false;
+                        break;
+                }
+
+                return ret_value;
+            });
+    ASSERT_EQ(value, dec_value);
+    Cdr::state dec_state_end(cdr);
+    ASSERT_EQ(enc_state_end, dec_state_end);
+    //}
+}
+
+/*!
+ * @test Test the decoder can get the XCDR version from the encapsulation.
+ * @code{.idl}
+ * struct AutoSelectStruct
+ * {
+ *     @id(0)
+ *     unsigned short value1;
+ *     @id(1)
+ *     unsigned long value2;
+ * };
+ * @endcode
+ */
+TEST_P(XCdrv1Test, auto_selection_on_decode)
+{
+    EncodingAlgorithmFlag encoding = std::get<0>(GetParam());
+    Cdr::Endianness endianness = Cdr::Endianness::LITTLE_ENDIANNESS;
+    auto buffer =
+            std::unique_ptr<char, void (*)(
+        void*)>{reinterpret_cast<char*>(calloc(100, sizeof(char))), free};
+    FastBuffer fast_buffer(buffer.get(), 100);
+
+    //{ Encode a ushort and a ulong.
+    Cdr cdr(fast_buffer, endianness, CdrVersion::XCDRv1);
+    const uint16_t us {0x01FC};
+    const uint32_t ul {0x01FC1FCD};
+    cdr.set_encoding_flag(encoding);
+    cdr.serialize_encapsulation();
+    Cdr::state enc_state(cdr);
+    cdr.begin_serialize_type(enc_state, encoding);
+    cdr << MemberId(0) << us;
+    cdr << MemberId(1) << ul;
+    cdr.end_serialize_type(enc_state);
+    //}
+
+    //{ Decode a ushort and a ulong.
+    Cdr dcdr(fast_buffer, endianness, std::get<1>(GetParam()));
+    uint16_t dus{0};
+    uint32_t dul{0};
+    dcdr.read_encapsulation();
+    ASSERT_EQ(dcdr.get_encoding_flag(), encoding);
+    ASSERT_EQ(dcdr.endianness(), endianness);
+    dcdr.deserialize_type(encoding, [&](Cdr& dcdr_inner, const MemberId& mid) -> bool
+            {
+                bool ret_value = true;
+                switch (mid.id)
+                {
+                    case 0:
+                        dcdr_inner >> dus;
+                        break;
+                    case 1:
+                        dcdr_inner >> dul;
+                        break;
+                    default:
+                        ret_value = false;
+                        break;
+                }
+
+                return ret_value;
+            });
+    ASSERT_EQ(us, dus);
+    ASSERT_EQ(ul, dul);
+}
+
+/*!
+ * @test Test serialization of a octet member changing the XCdrHeaderSelection
```

Comment:
Typo:

```suggestion
 * @test Test serialization of an octet member changing the XCdrHeaderSelection
```

Please, fix also other occurrences.

Reply:
Done in 07380dc